### PR TITLE
Update and rewrite of na_ontap_user

### PR DESCRIPTION
**Pull request by carchi8py**
##### SUMMARY
- Rewrite of ontap_user to use NetAppModule
- Add option applications
- Unit tests. 

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME
- na_ontap_user

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Review by thedoubl3j — APPROVED** [review]
cleaned up and looks clean. added more to module utils and included unit tests, lgtm

**Comment by thedoubl3j** [comment]
@carchi8py mind rebasing this for us?

File: lib/ansible/modules/storage/netapp/na_ontap_user.py [hunk 1]
```diff
@@ -26,24 +26,24 @@
...
-  application:
+  applications:
     description:
-    - Application to grant access to.
+    - List of application to grant access to.
     required: true
+    type: list
     choices: ['console', 'http','ontapi','rsh','snmp','service-processor','sp','ssh','telnet']
-
+    version_added: "2.8"
```
Comment (gundalow) [inline lib/ansible/modules/storage/netapp/na_ontap_user.py]: Please remove this line.

**Review by gundalow — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/storage/netapp/na_ontap_user.py (hunk 1 above)
Comment (carchi8py) [reply lib/ansible/modules/storage/netapp/na_ontap_user.py]: removed.

**Review by gundalow — APPROVED** [review]

**Pull request merged**